src/layout/monitor.rs

```diff
@@ -575,6 +575,18 @@ impl<W: LayoutElement> Monitor<W> {
         self.clean_up_workspaces();
     }
 
+    // returns a rectangle describing where the focused tile is located in visual coordinates
+    pub fn active_tile_visual_rectangle(&mut self) -> Option<Rectangle<i32, Logical>> {
+        let workspace = self.active_workspace();
+        if workspace.columns.is_empty() {
+            return None;
+        }
+        let col = &mut workspace.columns[workspace.active_column_idx];
+        let tile_size = col.tiles[col.active_tile_idx].tile_size();
+        let tile_pos = workspace.active_tile_visual_pos();
```

Comment:
This should probably take the workspace switch gesture into account?

Reply:
I'm not quite sure what to change, to me it seems to work fine as is.